service/logs.go

```diff
@@ -14,36 +16,43 @@ type Log struct {
 	Error      io.ReadCloser
 }
 
-// Logs gives service's logs and applies dependencies filter to filter logs.
-// if dependencies has a length of zero all dependency logs will be provided.
+// Logs gives service's logs streams. when dependencies filter is not provided, it'll give
+// logs for all dependencies otherwise it'll only give logs for specified dependencies.
+// note that, service itself is also a dependency defined with special "service" key.
+// in order to get service's own logs, "service" key must be included to dependencies filter.
 func (s *Service) Logs(c container.Container, dependencies ...string) ([]*Log, error) {
 	var (
 		logs       []*Log
 		isNoFilter = len(dependencies) == 0
 	)
-	addLog := func(dep *Dependency, name string) error {
-		if isNoFilter || xstrings.SliceContains(dependencies, name) {
-			rstd, rerr, err := dep.Logs(c, s.namespace())
+	for _, d := range append(s.Dependencies, s.Configuration) {
+		// Service.Configuration can be nil so, here is a check for it.
+		if d == nil {
+			continue
+		}
+		if isNoFilter || xstrings.SliceContains(dependencies, d.Key) {
+			var r io.ReadCloser
+			r, err := c.ServiceLogs(d.namespace(s.namespace()))
+			if err != nil {
+				return nil, err
+			}
+			rstd, sw := io.Pipe()
+			rerr, ew := io.Pipe()
+			go func(dstout, dsterr io.Writer, r io.ReadCloser) {
+				if _, err := stdcopy.StdCopy(dstout, dsterr, r); err != nil {
+					r.Close()
+					logrus.Errorln(err)
+				}
+			}(sw, ew, r)
 			if err != nil {
-				return err
+				return nil, err
```

Comment:
There is no way that `if err != nil` will return true. This test can be removed.

Reply:
done